test/realm_set_test.dart

```diff
@@ -0,0 +1,535 @@
+      var config = Configuration.local([TestRealmSets.schema, Car.schema]);
+      var realm = getRealm(config);
+
+      var testSet = TestRealmSets(1);
+
+      realm.write(() {
+        realm.add(testSet);
+      });
+
+      expect(realm.find<TestRealmSets>(1), isNotNull);
+
+      testSet = realm.find<TestRealmSets>(1)!;
+      var set = testSet.setByType(type).set;
+
+      expect(set.length, equals(0));
+    });
+
+    test('RealmSet<$type> create from unmanaged', () {
+      var config = Configuration.local([TestRealmSets.schema, Car.schema]);
+      var realm = getRealm(config);
+
+      var testSet = TestRealmSets(1);
+      var set = testSet.setByType(type).set;
+      var values = testSet.values(type);
+
+      for (var value in values) {
+        set.add(value);
+      }
+
+      realm.write(() {
+        realm.add(testSet);
+      });
+
+      testSet = realm.find<TestRealmSets>(1)!;
+      set = testSet.setByType(type).set;
+      expect(set.length, equals(values.length));
+      values = testSet.getValuesOrManagedValues(realm, type);
+
+      for (var value in values) {
+        expect(set.contains(value), true);
+      }
+    });
+
+    test('RealmSet<$type> contains', () {
+      var config = Configuration.local([TestRealmSets.schema, Car.schema]);
+      var realm = getRealm(config);
+
+      var testSet = TestRealmSets(1);
+      var set = testSet.setByType(type).set;
+      var values = testSet.values(type);
+
+      realm.write(() {
+        realm.add(testSet);
+      });
+
+      set = testSet.setByType(type).set;
+
+      expect(set.contains(values.first), false);
+
+      realm.write(() {
+        set.add(values.first);
+      });
+
+      testSet = realm.find<TestRealmSets>(1)!;
+      set = testSet.setByType(type).set;
+      expect(set.contains(values.first), true);
+    });
+
+    test('RealmSet<$type> add', () {
+      var config = Configuration.local([TestRealmSets.schema, Car.schema]);
+      var realm = getRealm(config);
+
+      final testSet = TestRealmSets(1);
+
+      var values = testSet.values(type);
+
+      realm.write(() {
+        realm.add(testSet);
+        var set = testSet.setByType(type).set;
+        set.add(values.first);
+      });
+
+      var set = testSet.setByType(type).set;
+      // values = testSet.values(type);
+      values = testSet.getValuesOrManagedValues(realm, type);
+
+      expect(set.contains(values.first), true);
+    });
+
+    test('RealmSet<$type> remove', () {
+      var config = Configuration.local([TestRealmSets.schema, Car.schema]);
+      var realm = getRealm(config);
+
+      var testSet = TestRealmSets(1);
+
+      realm.write(() {
+        realm.add(testSet);
+      });
+
+      var set = testSet.setByType(type).set;
+      var values = testSet.values(type);
+
+      realm.write(() {
+        set.add(values.first);
+      });
+
+      expect(set.length, 1);
+
+      realm.write(() {
+        expect(set.remove(values.first), true);
```

Comment:
This should probably also test the case where we remove a value that's missing from the set.

Reply:
fixed